packages/esm-patient-orders-app/src/lab-results/lab-results-form-field.component.tsx

```diff
@@ -15,13 +15,34 @@ interface ResultFormFieldProps {
 const ResultFormField: React.FC<ResultFormFieldProps> = ({ concept, control, defaultValue, errors }) => {
   const { t } = useTranslation();
 
-  const printValueRange = (concept: LabOrderConcept) => {
-    if (concept?.datatype?.display === 'Numeric') {
-      const maxVal = Math.max(concept?.hiAbsolute, concept?.hiCritical, concept?.hiNormal);
-      const minVal = Math.min(concept?.lowAbsolute, concept?.lowCritical, concept?.lowNormal);
-      return ` (${minVal ?? 0} - ${maxVal > 0 ? maxVal : '--'} ${concept?.units ?? ''})`;
+  // TODO: Reference ranges should be dynamically adjusted based on patient demographics:
+  // - Age-specific ranges (e.g., pediatric vs adult values)
+  // - Gender-specific ranges where applicable
+  const formatLabRange = (concept: LabOrderConcept) => {
+    const {
+      datatype: { hl7Abbreviation },
+    } = concept ?? {};
+    if (hl7Abbreviation !== 'NM') return '';
+
+    const { hiAbsolute, lowAbsolute, units } = concept;
+    const displayUnit = units ? ` ${units}` : '';
+
+    const hasLowerLimit = lowAbsolute !== null && lowAbsolute !== undefined;
+    const hasUpperLimit = hiAbsolute !== null && hiAbsolute !== undefined;
+
+    if (hasLowerLimit && hasUpperLimit) {
+      return ` (${lowAbsolute} - ${hiAbsolute}) ${displayUnit}`;
+    }
+
+    if (hasUpperLimit) {
+      return ` (<= ${hiAbsolute}) ${displayUnit}`;
+    }
+
+    if (hasLowerLimit) {
+      return ` (>= ${lowAbsolute}) ${displayUnit}`;
     }
```

Comment:
While not functionally different, this makes the logic a little clearer:

```suggestion
    if (hasLowerLimit && hasUpperLimit) {
      return ` (${lowAbsolute} - ${hiAbsolute}) ${displayUnit}`;
    } else if (hasUpperLimit) {
      return ` (<= ${hiAbsolute}) ${displayUnit}`;
    } else if (hasLowerLimit) {
      return ` (>= ${lowAbsolute}) ${displayUnit}`;
    }
```